lib/elements/tab_expand_controller.dart

```diff
@@ -0,0 +1,209 @@
+import 'dart:async';
+import 'dart:html';
+
+import 'package:meta/meta.dart';
+import 'package:split/split.dart';
+
+import 'button.dart';
+import 'counter.dart';
+import 'elements.dart';
+
+enum TabState {
+  closed,
+  ui,
+  docs,
+  console,
+}
+
+/// Manages the bottom-left panel and tabs
+class TabExpandController {
+  final MDCButton uiOutputButton;
+  final MDCButton consoleButton;
+  final MDCButton docsButton;
+  final MDCButton closeButton;
+  final DElement console;
+  final DElement docs;
+  final Counter unreadCounter;
+  final DElement iframe;
+
+  /// The element to give the top half of the split when this panel
+  /// opens
+  final Element topSplit;
+
+  /// The element to give the bottom half of the split
+  final Element bottomSplit;
+
+  final List<StreamSubscription> _subscriptions = [];
+
+  TabState _state;
+  Splitter _splitter;
+  bool _splitterConfigured = false;
+
+  TabState get state => _state;
+
+  TabExpandController({
+    this.uiOutputButton,
+    @required this.consoleButton,
+    @required this.docsButton,
+    @required this.closeButton,
+    IFrameElement iframeElement,
+    @required Element consoleElement,
+    @required Element docsElement,
+    @required this.topSplit,
+    @required this.bottomSplit,
+    @required this.unreadCounter,
+  })  : console = DElement(consoleElement),
+        docs = DElement(docsElement),
+        iframe = iframeElement == null ? null : DElement(iframeElement) {
+    _state = TabState.closed;
+    iframe?.setAttr('hidden');
+    console.setAttr('hidden');
+    docs.setAttr('hidden');
+
+    if (uiOutputButton != null) {
+      _subscriptions.add(uiOutputButton.onClick.listen((_) {
+        toggleIframe();
+      }));
+    }
+
+    _subscriptions.add(consoleButton.onClick.listen((_) {
+      toggleConsole();
+    }));
+
+    _subscriptions.add(docsButton.onClick.listen((_) {
+      toggleDocs();
+    }));
+
+    _subscriptions.add(closeButton.onClick.listen((_) {
+      _hidePanel();
+    }));
+  }
+
+  void toggleIframe() {
+    if (_state == TabState.closed) {
```

Comment:
Since this is an enum, if you use a switch instead of if/elses, you get the benefit of analysis which says whether you've missed a case or not.